src/dataflow-types/src/types.rs

```diff
@@ -1162,6 +1162,13 @@ pub struct KafkaSinkConnectorBuilder {
     pub reuse_topic: bool,
     // Source dependencies for exactly-once sinks.
     pub transitive_source_dependencies: Vec<GlobalId>,
+    pub retention: KafkaSinkConnectorRetention,
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct KafkaSinkConnectorRetention {
+    pub retention_ms: Option<i64>,
+    pub retention_bytes: Option<i64>,
```

Comment:
I chose to make these `Option`s rather than injecting a default value here.  Let me know if you think I should convert these to `i64`s.